starlark-test/src/lib.rs

```diff
@@ -108,54 +90,41 @@ def assert_(cond, msg="assertion failed"):
     )
     .unwrap();
     prelude.freeze();
-    for (offset, content) in read_input(path) {
-        let err = if let Some(x) = content.find("###") {
-            let err = content.get(x + 3..).unwrap().trim();
-            err.get(..err.find('\n').unwrap_or_else(|| err.len()))
-                .unwrap()
-        } else {
-            ""
-        };
-        let content = std::iter::repeat("\n").take(offset).collect::<String>() + &content;
-        match eval(
-            &map,
-            &path,
-            &content,
-            starlark::syntax::dialect::Dialect::Bzl,
-            &mut prelude.child(&path),
-            global.clone(),
-        ) {
-            Err(p) => {
-                if err.is_empty() {
-                    Emitter::stderr(ColorConfig::Always, Some(&map.lock().unwrap())).emit(&[p]);
-                    return false;
-                } else if !assert_diagnostic(p, err, path, offset, &map) {
-                    return false;
-                }
+    let err = if let Some(x) = content.find("###") {
+        let err = content.get(x + 3..).unwrap().trim();
+        err.get(..err.find('\n').unwrap_or_else(|| err.len()))
+            .unwrap()
+    } else {
+        ""
+    };
+    match eval(
+        &map,
+        path,
+        &content,
+        starlark::syntax::dialect::Dialect::Bzl,
+        &mut prelude.child(path),
+        global.clone(),
+    ) {
+        Err(p) => {
+            if err.is_empty() {
+                Emitter::stderr(ColorConfig::Always, Some(&map.lock().unwrap())).emit(&[p]);
+                panic!();
+            } else if !assert_diagnostic(p, err, "test", 0, &map) {
```

Comment:
We are loosing the offset here, which means we will not see the actual line number anymore. Can you pass the offset as an argument to do_conformance_test and reinject it from the code generation?

Reply:
@damienmg we are not losing offset, newlines are in generated files: https://gist.github.com/stepancheg/c68a83f8a0e5309ad8cec8ad4283658a#file-rust-testcases-rs-L39

Reply:
OK, I see. Give me a minute.